From 6311ef7d3f2283766f4cb6a5dd8841ba6eeb5ddf Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 24 Apr 2025 17:52:09 +0200
Subject: [PATCH 1692/1695] dts: stm32: change scmi for CM33 shared memory
 address on stm32mp23 soc

Use a dedicated memory region for the shared memory used for SCMI commands
between CA35_S an CM33_NS.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 236584c910e48ac32bdfbee6bc2a97ae284a6a37 dts: stm32: change scmi for CM33 shared memory address on stm32mp23 soc]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I547b3394986dc1ad5843a42c2a9f903ed20d5009
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/453945
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp23-st-scmi-cfg.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp23-st-scmi-cfg.dtsi b/core/arch/arm/dts/stm32mp23-st-scmi-cfg.dtsi
index 7f763ffcc..0908ac5c9 100644
--- a/core/arch/arm/dts/stm32mp23-st-scmi-cfg.dtsi
+++ b/core/arch/arm/dts/stm32mp23-st-scmi-cfg.dtsi
@@ -922,7 +922,7 @@
 			scmi-channel-id = <0>;
 			mbox-names = "scmi_cm33";
 			mboxes = <&ipcc1 RIF_IPCC_CPU1_CHANNEL(16)>;
-			shmem = <&ipc_shmem>;
+			shmem = <&scmi_cid2_ns>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-- 
2.47.2

